### ISSUE TITLE
Moving away from compiled type metadata and use Libraries that create an schema and type

### ISSUE THREAD

**Issue by M-jerez** [MionKit/mion#75]
Seem that the Typescript Community and ten Typescript team itself favors Libraries that create an schema and later you can get a type from it. ie, zod, valibot etc, rather than libraries that att metadata to types at ryntime. 

More info here: https://github.com/akutruff/typescript-needs-types. mentioned repo is trying for the Typescript team to include types at runtime but from some comments from the typescript team, that never gonna happen. 

Also using a typescript transformer can make things more complicated when using new JS runtimes like Deno and Bun, that includes an standard typescript compiler. 

So to summarize, look like the community prefers  schema libraries like zod rather than add metadata for types at runtime. 

- [ ]  explore using libraries like zod or valibot and make it configurable

**Comment by M-jerez** [comment]
we are using or own run-types library
